In [7]:
import pandas as pd
import numpy as np
from sklearn.cluster import KMeans
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import MinMaxScaler
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeClassifier
from sklearn import metrics
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler
from sklearn.neural_network import MLPClassifier
from sklearn.metrics import classification_report,confusion_matrix
from sklearn.metrics import accuracy_score
from sklearn.tree import DecisionTreeClassifier
from sklearn import metrics
from sklearn.tree import export_graphviz
from IPython.display import Image  
import pydotplus
from io import StringIO
from sklearn.metrics import mean_squared_error, r2_score
import time

In [8]:
df=pd.read_csv('AllSensor.csv',names=['Number','Sensor_Address','Time','Date','Temperature','Humidity','Light','Motion','CO2','VDD'])
#df=pd.DataFrame([df],columns=['Number','Sensor Address','Time','Date','Temperature','Humidity','Light','Motion','CO2','VDD'])
weather=pd.read_csv('weather.csv',names=['Number','Temp_weather','Temperature_Max','Temperature_min','Pressure','Humidity_weather','Wind_speed','Sunset_time','Sunrise_time','Weather','Latitude','Longitude','Time','Date'])
#weather.head(1)
result_matrix = pd.DataFrame(columns=['Name','R_Score','Linear_regression_time','Linear_regression_CPU_time',
                                      'NN','Neural_Network_time','Neural_Network_CPU_time','DT',
                                      'Decision_Tree_time','Decision_Tree_CPU_time'])

In [9]:
weather = weather.drop(['Number','Latitude','Longitude'],axis = 1)
weather['DateTime'] = weather['Date'] + ' ' + weather['Time']
weather['Time'] = pd.to_datetime(weather['Time'], format='%H:%M:%S').dt.time
weather['Date'] = pd.to_datetime(weather['Date'], format='%Y-%m-%d').dt.date
weather['DateTime'] = pd.to_datetime(weather['DateTime'], format='%Y-%m-%d %H:%M:%S')
weather['Minute']=weather.DateTime.dt.minute
weather['Hour']=weather.DateTime.dt.hour
W={'Clear':1,'Clouds':2,'Drizzle':3,'Fog':4,'Haze':5,'Mist':6,'Rain':7,'Snow':8,'Thunderstorm':9}
weather.Weather = [W[i] for i in weather.Weather] 


In [10]:
df.dropna
df['DateTime'] = df['Date'] + ' ' + df['Time']
df = df.drop(['Number'], axis =1)
df['Time'] = pd.to_datetime(df['Time'], format='%H:%M:%S',errors='coerce').dt.time
df['Date'] = pd.to_datetime(df['Date'], format='%Y-%m-%d',errors='coerce').dt.date
df.dropna
df['DateTime'] = pd.to_datetime(df['DateTime'], format='%Y-%m-%d %H:%M:%S',errors='coerce')
df.dropna
df['Minute']=df.DateTime.dt.minute
df['Hour']=df.DateTime.dt.hour

In [12]:
s= {
    'a8-17-58-ff-fe-03-64-cb',
    'a8-17-58-ff-fe-03-65-48',
    'a8-17-58-ff-fe-03-65-49',
    'a8-17-58-ff-fe-03-65-4a',
    'a8-17-58-ff-fe-03-65-4b',
    'a8-17-58-ff-fe-03-65-4c',
    'a8-17-58-ff-fe-03-65-4d',
    'a8-17-58-ff-fe-03-66-ac',
    'a8-17-58-ff-fe-03-66-af',
    'a8-17-58-ff-fe-03-66-b6',
    'a8-17-58-ff-fe-03-66-b7',
    'a8-17-58-ff-fe-03-66-b8',
    'a8-17-58-ff-fe-03-66-b9',
    'a8-17-58-ff-fe-03-66-ba',
    'a8-17-58-ff-fe-03-67-1a',
    'a8-17-58-ff-fe-03-93-8e',
    'a8-17-58-ff-fe-03-93-8f',
    'a8-17-58-ff-fe-03-93-90',
    'a8-17-58-ff-fe-03-93-91',
    'a8-17-58-ff-fe-03-93-92',
    'a8-17-58-ff-fe-03-93-93',
    'a8-17-58-ff-fe-03-93-97',
    'a8-17-58-ff-fe-03-93-d0',
    'a8-17-58-ff-fe-03-93-d1',
    'a8-17-58-ff-fe-03-93-d2',
    'a8-17-58-ff-fe-03-93-d3',
    'a8-17-58-ff-fe-03-93-d4',
    'a8-17-58-ff-fe-03-93-d7'
    #'a8-17-58-ff-fe-03-93-d8',
    'a8-17-58-ff-fe-03-93-d9',
    'a8-17-58-ff-fe-03-93-da'}
for i in s:
    print (i)

a8-17-58-ff-fe-03-65-4a
a8-17-58-ff-fe-03-66-ba
a8-17-58-ff-fe-03-66-b8
a8-17-58-ff-fe-03-65-4b
a8-17-58-ff-fe-03-65-48
a8-17-58-ff-fe-03-93-d4
a8-17-58-ff-fe-03-65-4d
a8-17-58-ff-fe-03-93-91
a8-17-58-ff-fe-03-66-af
a8-17-58-ff-fe-03-93-97
a8-17-58-ff-fe-03-93-8e
a8-17-58-ff-fe-03-93-d2
a8-17-58-ff-fe-03-67-1a
a8-17-58-ff-fe-03-66-ac
a8-17-58-ff-fe-03-93-93
a8-17-58-ff-fe-03-93-d1
a8-17-58-ff-fe-03-93-d3
a8-17-58-ff-fe-03-64-cb
a8-17-58-ff-fe-03-66-b9
a8-17-58-ff-fe-03-93-d7a8-17-58-ff-fe-03-93-d9
a8-17-58-ff-fe-03-66-b7
a8-17-58-ff-fe-03-93-da
a8-17-58-ff-fe-03-66-b6
a8-17-58-ff-fe-03-93-90
a8-17-58-ff-fe-03-93-92
a8-17-58-ff-fe-03-65-4c
a8-17-58-ff-fe-03-65-49
a8-17-58-ff-fe-03-93-8f
a8-17-58-ff-fe-03-93-d0


In [ ]:
# 'a8-17-58-ff-fe-03-67-1a':'Home1_Bedroom','a8-17-58-ff-fe-03-66-b9':'Home3_Bedroom',
#    'a8-17-58-ff-fe-03-65-48':'Home4_Bedroom','a8-17-58-ff-fe-03-66-b7':'Home5_Bedroom',
#    'a8-17-58-ff-fe-03-66-ac':'Home6_Bedroom','a8-17-58-ff-fe-03-65-4c':'Home7_Bedroom',
#    'a8-17-58-ff-fe-03-64-cb':'Home8_Bedroom','a8-17-58-ff-fe-03-66-ba':'Home11_Bedroom',
#    'a8-17-58-ff-fe-03-93-8e':'Home12_Bedroom','a8-17-58-ff-fe-03-93-90':'Home13_Bedroom',
#    'a8-17-58-ff-fe-03-93-92':'Home14_Bedroom','a8-17-58-ff-fe-03-93-97':'Home16_Bedroom',
#    'a8-17-58-ff-fe-03-93-d2':'Home20_Bedroom','a8-17-58-ff-fe-03-93-d4':'Home21_Bedroom',
#    'a8-17-58-ff-fe-03-93-d9':'Home24_Bedroom','a8-17-58-ff-fe-03-66-af':'Home1_Livingroom',
#    'a8-17-58-ff-fe-03-66-b6':'Home3_Livingroom',
#    'a8-17-58-ff-fe-03-65-49':'Home4_Livingroom','a8-17-58-ff-fe-03-66-b8':'Home5_Livingroom',
#s= {
#    'a8-17-58-ff-fe-03-65-4b':'Home6_Livingroom','a8-17-58-ff-fe-03-65-4d':'Home7_Livingroom',
#    'a8-17-58-ff-fe-03-64-cb':'Home8_Livingroom','a8-17-58-ff-fe-03-65-49':'Home11_Livingroom',
#    'a8-17-58-ff-fe-03-93-8f':'Home12_Livingroom',
    #'a8-17-58-ff-fe-03-93-91':'Home13_Livingroom',
    #'a8-17-58-ff-fe-03-93-93':'Home14_Livingroom','a8-17-58-ff-fe-03-93-d0':'Home18_Livingroom',
    #'a8-17-58-ff-fe-03-93-d1':'Home19_Livingroom','a8-17-58-ff-fe-03-93-d3':'Home20_Livingroom',
    #'a8-17-58-ff-fe-03-93-d7':'Home22_Livingroom',
    #'a8-17-58-ff-fe-03-93-d8':'Home23_Livingroom',
 #   'a8-17-58-ff-fe-03-93-da':'Home24_Livingroom',
 #   'a8-17-58-ff-fe-03-66-ba':'Home2_Livingroom','a8-17-58-ff-fe-03-65-4a':'Unknown1'}
for i in s:
    df1 = df[df['Sensor_Address']== i ]
    df1.dropna
    #df1['DateTime'] = df1['Date'] + ' ' + df1['Time']
    #df1 = df1.drop(['Number'], axis =1)
    #df1['Time'] = pd.to_datetime(df1['Time'], format='%H:%M:%S',errors='coerce').dt.time
    #df1['Date'] = pd.to_datetime(df1['Date'], format='%Y-%m-%d',errors='coerce').dt.date
    #df1.dropna
    #df1['DateTime'] = pd.to_datetime(df1['DateTime'], format='%Y-%m-%d %H:%M:%S',errors='coerce')
    #df1.dropna
    #df1['Minute']=df1.DateTime.dt.minute
    #df1['Hour']=df1.DateTime.dt.hour
    result = pd.merge(weather, df1, on=['Date','Hour'])
    result = result.reset_index()
    X = pd.DataFrame(result,columns=['Temp_weather','Pressure','Humidity_weather','Wind_speed','Weather','Hour'])
    y = result['Temperature'].round().astype(int)
    X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=0.3, random_state=0)
    lr = time.time()
    clr = time.clock()
    Model = LinearRegression()
    Model.fit(X_train, y_train)
    lr = time.time() - lr
    clr = time.clock() - clr
    Model.score(X_train, y_train)
    y_pred = Model.predict(X_test)
    rscore = r2_score(y_test, y_pred)
    print ('rscore is', rscore)
    scaler = StandardScaler()
    scaler.fit(X_train)
    StandardScaler(copy=True, with_mean=True, with_std=True)
    X_train1 = scaler.transform(X_train)
    X_test1 = scaler.transform(X_test)
    nn = time.time()
    cnn = time.clock()
    mlp = MLPClassifier(hidden_layer_sizes=(6,12,12,12,12,12,12,12,12,12,12,12,12,6), activation='relu', solver='adam', max_iter=500)
    mlp.fit(X_train1,y_train)
    nn = time.time() - nn
    cnn = time.clock() - cnn
    predictions = mlp.predict(X_test1)
    NN = accuracy_score(y_test, predictions)
    print ('NN accuracy', NN)
    dt = time.time()
    cdt = time.clock()
    clf = DecisionTreeClassifier()
    # Train Decision Tree Classifer
    clf = clf.fit(X_train,y_train)
    dt = time.time() - dt
    cdt = time.clock() - cdt
    y_pred = clf.predict(X_test)
    DT = metrics.accuracy_score(y_test, y_pred)
    print ('DT accuracy', DT)
    result_matrix = result_matrix.append({'Name': i,'R_Score':rscore,'Linear_regression_time':lr,
                                          'Linear_regression_CPU_time':clr,
                                          'NN':NN,'Neural_Network_time':nn,'Neural_Network_CPU_time':cnn,
                                          'DT':DT,'Decision_Tree_time':dt,'Decision_Tree_CPU_time':cdt
                                         }, ignore_index=True)
    print (result_matrix)
    result = pd.DataFrame(None)
    df1 = pd.DataFrame(None)
    

/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.44616488862201786


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:59: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


NN accuracy 0.7355961601208978


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:64: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:69: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


DT accuracy 0.9639023887619996
                      Name   R_Score  Linear_regression_time  \
0  a8-17-58-ff-fe-03-65-4a  0.446165                0.037873   

   Linear_regression_CPU_time        NN  Neural_Network_time  \
0                    0.049301  0.735596          2026.338197   

   Neural_Network_CPU_time        DT  Decision_Tree_time  \
0              2025.366948  0.963902            0.651157   

   Decision_Tree_CPU_time  
0                0.653132  


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.20037141022974136


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:59: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


NN accuracy 0.732756930984167


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:64: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:69: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


DT accuracy 0.9535756546231969
                      Name   R_Score  Linear_regression_time  \
0  a8-17-58-ff-fe-03-65-4a  0.446165                0.037873   
1  a8-17-58-ff-fe-03-66-ba  0.200371                0.046867   

   Linear_regression_CPU_time        NN  Neural_Network_time  \
0                    0.049301  0.735596          2026.338197   
1                    0.060889  0.732757          2372.366788   

   Neural_Network_CPU_time        DT  Decision_Tree_time  \
0              2025.366948  0.963902            0.651157   
1              2371.883254  0.953576            0.709361   

   Decision_Tree_CPU_time  
0                0.653132  
1                0.711811  


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.5204367007707087


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:59: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


NN accuracy 0.7139468417721114


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:64: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:69: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


DT accuracy 0.9726408398923281
                      Name   R_Score  Linear_regression_time  \
0  a8-17-58-ff-fe-03-65-4a  0.446165                0.037873   
1  a8-17-58-ff-fe-03-66-ba  0.200371                0.046867   
2  a8-17-58-ff-fe-03-66-b8  0.520437                0.048600   

   Linear_regression_CPU_time        NN  Neural_Network_time  \
0                    0.049301  0.735596          2026.338197   
1                    0.060889  0.732757          2372.366788   
2                    0.061989  0.713947          2409.425981   

   Neural_Network_CPU_time        DT  Decision_Tree_time  \
0              2025.366948  0.963902            0.651157   
1              2371.883254  0.953576            0.709361   
2              2409.211939  0.972641            0.822080   

   Decision_Tree_CPU_time  
0                0.653132  
1                0.711811  
2                0.824466  


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.6586284936804436


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:59: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


NN accuracy 0.6179219671113869


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:64: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:69: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


DT accuracy 0.9216309856241597
                      Name   R_Score  Linear_regression_time  \
0  a8-17-58-ff-fe-03-65-4a  0.446165                0.037873   
1  a8-17-58-ff-fe-03-66-ba  0.200371                0.046867   
2  a8-17-58-ff-fe-03-66-b8  0.520437                0.048600   
3  a8-17-58-ff-fe-03-65-4b  0.658628                0.053819   

   Linear_regression_CPU_time        NN  Neural_Network_time  \
0                    0.049301  0.735596          2026.338197   
1                    0.060889  0.732757          2372.366788   
2                    0.061989  0.713947          2409.425981   
3                    0.070360  0.617922          2275.351431   

   Neural_Network_CPU_time        DT  Decision_Tree_time  \
0              2025.366948  0.963902            0.651157   
1              2371.883254  0.953576            0.709361   
2              2409.211939  0.972641            0.822080   
3              2275.160283  0.921631            0.884058   

   Decision_Tree_CPU_time 

/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.08632305921651684


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:59: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


NN accuracy 0.6551393444195102


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:64: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:69: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


DT accuracy 0.8799508985940887
                      Name   R_Score  Linear_regression_time  \
0  a8-17-58-ff-fe-03-65-4a  0.446165                0.037873   
1  a8-17-58-ff-fe-03-66-ba  0.200371                0.046867   
2  a8-17-58-ff-fe-03-66-b8  0.520437                0.048600   
3  a8-17-58-ff-fe-03-65-4b  0.658628                0.053819   
4  a8-17-58-ff-fe-03-65-48  0.086323                0.016177   

   Linear_regression_CPU_time        NN  Neural_Network_time  \
0                    0.049301  0.735596          2026.338197   
1                    0.060889  0.732757          2372.366788   
2                    0.061989  0.713947          2409.425981   
3                    0.070360  0.617922          2275.351431   
4                    0.020685  0.655139           174.157434   

   Neural_Network_CPU_time        DT  Decision_Tree_time  \
0              2025.366948  0.963902            0.651157   
1              2371.883254  0.953576            0.709361   
2              2409

/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.5451298024275921


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:59: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


NN accuracy 0.7241242396092415


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:64: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:69: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


DT accuracy 0.9362470408438465
                      Name   R_Score  Linear_regression_time  \
0  a8-17-58-ff-fe-03-65-4a  0.446165                0.037873   
1  a8-17-58-ff-fe-03-66-ba  0.200371                0.046867   
2  a8-17-58-ff-fe-03-66-b8  0.520437                0.048600   
3  a8-17-58-ff-fe-03-65-4b  0.658628                0.053819   
4  a8-17-58-ff-fe-03-65-48  0.086323                0.016177   
5  a8-17-58-ff-fe-03-93-d4  0.545130                0.049490   

   Linear_regression_CPU_time        NN  Neural_Network_time  \
0                    0.049301  0.735596          2026.338197   
1                    0.060889  0.732757          2372.366788   
2                    0.061989  0.713947          2409.425981   
3                    0.070360  0.617922          2275.351431   
4                    0.020685  0.655139           174.157434   
5                    0.063496  0.724124          2314.325357   

   Neural_Network_CPU_time        DT  Decision_Tree_time  \
0          

/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.44779944111058134


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:59: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


NN accuracy 0.6400527088137522


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:64: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:69: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


DT accuracy 0.9438920667245665
                      Name   R_Score  Linear_regression_time  \
0  a8-17-58-ff-fe-03-65-4a  0.446165                0.037873   
1  a8-17-58-ff-fe-03-66-ba  0.200371                0.046867   
2  a8-17-58-ff-fe-03-66-b8  0.520437                0.048600   
3  a8-17-58-ff-fe-03-65-4b  0.658628                0.053819   
4  a8-17-58-ff-fe-03-65-48  0.086323                0.016177   
5  a8-17-58-ff-fe-03-93-d4  0.545130                0.049490   
6  a8-17-58-ff-fe-03-65-4d  0.447799                0.021834   

   Linear_regression_CPU_time        NN  Neural_Network_time  \
0                    0.049301  0.735596          2026.338197   
1                    0.060889  0.732757          2372.366788   
2                    0.061989  0.713947          2409.425981   
3                    0.070360  0.617922          2275.351431   
4                    0.020685  0.655139           174.157434   
5                    0.063496  0.724124          2314.325357   
6       

/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.32600458181139147


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:59: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


NN accuracy 0.6790445506487439


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:64: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:69: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


DT accuracy 0.8449633530276337
                      Name   R_Score  Linear_regression_time  \
0  a8-17-58-ff-fe-03-65-4a  0.446165                0.037873   
1  a8-17-58-ff-fe-03-66-ba  0.200371                0.046867   
2  a8-17-58-ff-fe-03-66-b8  0.520437                0.048600   
3  a8-17-58-ff-fe-03-65-4b  0.658628                0.053819   
4  a8-17-58-ff-fe-03-65-48  0.086323                0.016177   
5  a8-17-58-ff-fe-03-93-d4  0.545130                0.049490   
6  a8-17-58-ff-fe-03-65-4d  0.447799                0.021834   
7  a8-17-58-ff-fe-03-93-91  0.326005                0.031230   

   Linear_regression_CPU_time        NN  Neural_Network_time  \
0                    0.049301  0.735596          2026.338197   
1                    0.060889  0.732757          2372.366788   
2                    0.061989  0.713947          2409.425981   
3                    0.070360  0.617922          2275.351431   
4                    0.020685  0.655139           174.157434   
5       

/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.36037175936212673


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:59: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


NN accuracy 0.6168530785101531


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:64: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:69: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


DT accuracy 0.9090672168530786
                      Name   R_Score  Linear_regression_time  \
0  a8-17-58-ff-fe-03-65-4a  0.446165                0.037873   
1  a8-17-58-ff-fe-03-66-ba  0.200371                0.046867   
2  a8-17-58-ff-fe-03-66-b8  0.520437                0.048600   
3  a8-17-58-ff-fe-03-65-4b  0.658628                0.053819   
4  a8-17-58-ff-fe-03-65-48  0.086323                0.016177   
5  a8-17-58-ff-fe-03-93-d4  0.545130                0.049490   
6  a8-17-58-ff-fe-03-65-4d  0.447799                0.021834   
7  a8-17-58-ff-fe-03-93-91  0.326005                0.031230   
8  a8-17-58-ff-fe-03-66-af  0.360372                0.044311   

   Linear_regression_CPU_time        NN  Neural_Network_time  \
0                    0.049301  0.735596          2026.338197   
1                    0.060889  0.732757          2372.366788   
2                    0.061989  0.713947          2409.425981   
3                    0.070360  0.617922          2275.351431   
4       

/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.602336161571691


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:59: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


NN accuracy 0.7851544715447154


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:64: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:69: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


DT accuracy 0.9526341463414634
                      Name   R_Score  Linear_regression_time  \
0  a8-17-58-ff-fe-03-65-4a  0.446165                0.037873   
1  a8-17-58-ff-fe-03-66-ba  0.200371                0.046867   
2  a8-17-58-ff-fe-03-66-b8  0.520437                0.048600   
3  a8-17-58-ff-fe-03-65-4b  0.658628                0.053819   
4  a8-17-58-ff-fe-03-65-48  0.086323                0.016177   
5  a8-17-58-ff-fe-03-93-d4  0.545130                0.049490   
6  a8-17-58-ff-fe-03-65-4d  0.447799                0.021834   
7  a8-17-58-ff-fe-03-93-91  0.326005                0.031230   
8  a8-17-58-ff-fe-03-66-af  0.360372                0.044311   
9  a8-17-58-ff-fe-03-93-97  0.602336                0.020446   

   Linear_regression_CPU_time        NN  Neural_Network_time  \
0                    0.049301  0.735596          2026.338197   
1                    0.060889  0.732757          2372.366788   
2                    0.061989  0.713947          2409.425981   
3       

/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:40: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:44: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


rscore is 0.5747830463114021


/srv/conda/envs/notebook/lib/python3.7/site-packages/ipykernel_launcher.py:55: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead


In [ ]:
result_matrix


feature_cols = ['Temp_weather','Pressure','Humidity_weather','Wind_speed','Weather','Hour']

dot_data = StringIO()
export_graphviz(clf, out_file=dot_data,  
                filled=True, rounded=True,
                special_characters=True, feature_names = feature_cols,class_names=['20','21','22','23','24','25','26','27','28','29'])
graph = pydotplus.graph_from_dot_data(dot_data.getvalue())  
graph.write_png('diabetes.png')
Image(graph.create_png())


clf1 = DecisionTreeClassifier(criterion="entropy", max_depth=3)

# Train Decision Tree Classifer
clf1 = clf1.fit(X_train,y_train)

#Predict the response for test dataset
y_pred1 = clf1.predict(X_test)

# Model Accuracy, how often is the classifier correct?
metrics.accuracy_score(y_test, y_pred1)

dot_data = StringIO()
export_graphviz(clf1, out_file=dot_data,  
                filled=True, rounded=True,
                special_characters=True, feature_names = feature_cols,class_names=['20','21','22','23','24','25','26','27','28','29'])
graph = pydotplus.graph_from_dot_data(dot_data.getvalue())  
graph.write_png('tree.png')
Image(graph.create_png())

In [ ]:
#df3 = df2.groupby(['Date'])
#df3 = df3['Temperature'].mean()
#df3 = df3.reset_index()

In [ ]:
#plt.scatter(df3['Date'],df3['Temperature'])

In [ ]:
#weather1 = weather.groupby(['Date','Hour'])
#weather1 = weather1['Temperature','Temperature_Max','Temperature_min','Pressure','Humidity_weather','Wind_speed'].mean()

In [ ]:
#weather1 = weather1.reset_index()

 Name   R_Score        NN        DT
0   a8-17-58-ff-fe-03-66-ac  0.595430  0.744994  0.943777
1   a8-17-58-ff-fe-03-65-4c  0.468339  0.694472  0.954318
2   a8-17-58-ff-fe-03-64-cb  0.146548  0.713001  0.954878
3   a8-17-58-ff-fe-03-66-ba  0.200371  0.683995  0.953583
4   a8-17-58-ff-fe-03-93-90  0.297189  0.706077  0.864526
5   a8-17-58-ff-fe-03-93-92  0.447549  0.919539  0.985631
6   a8-17-58-ff-fe-03-93-97  0.602336  0.754309  0.952748
7   a8-17-58-ff-fe-03-93-d2  0.499422  0.712141  0.963377
8   a8-17-58-ff-fe-03-93-d4  0.545130  0.722693  0.936247
9   a8-17-58-ff-fe-03-93-d9  0.507864  0.711767  0.950648
10  a8-17-58-ff-fe-03-66-af  0.360372  0.594911  0.909053
11  a8-17-58-ff-fe-03-66-b6  0.216660  0.825937  0.869008
12  a8-17-58-ff-fe-03-66-b8  0.520437  0.752808  0.972641
13  a8-17-58-ff-fe-03-65-4b  0.658628  0.666369  0.921631

In [ ]:
#result = pd.merge(weather, df1, on=['Date','Time','DateTime','Minute','Hour'])

In [ ]:
#result.head()